database/migrations/2016_09_24_095751_create_stats_table.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateStatsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('clicks', function(Blueprint $table) {
+            $table->engine = 'InnoDB';
+
+            $table->timestamp('date');
+            $table->string('ip');
+            $table->string('country')->nullable();
+            $table->string('referer')->nullable();
+            $table->string('referer_host')->nullable();
+            $table->unsignedInteger('link_id');
+
+            $table->index('ip');
+            $table->index('referer');
+
+            $table->foreign('link_id')->references('id')->on('links')->onDelete('cascade');
+        });
+
+        Schema::table('links', function ($table) {
```

Comment:
Right. If the environmental variable for additional stats is enabled, then we should use the clicks table for additional data. Otherwise, we should not store data in the clicks table or pull from it.